src/components/tooltip/tooltip.js

```diff
@@ -128,6 +128,19 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
 
       var ngWindow = angular.element($window);
 
+      if (parent[0] && 'MutationObserver' in $window) {
+        // use an mutationObserver to tackle #2602
+        var observer = new MutationObserver(function(mutations) {
```

Comment:
If `observer` is going to be referenced later in the `"$destroy"` handler, it should have a more specific name.